src/tif_model_iterator/tif_kernel_iterator.py

```diff
@@ -174,6 +172,7 @@ def _create_pixel_coordinate_dataframe(
             [y for y in range(0, data.shape[2])] for x in range(0, data.shape[1])
         ]
 
+        # We have to have a square so we are getting the upper left, upper right, lower left and lower right pixels.
```

Comment:
No we don't? And for a random shape if you take upper left, upp right, lower left and lower right pixels, you don't always get a square, but you do always get a rectangle.

Reply:
These are the coordinates of the pixel corners  of satellite images, idd i assume here that the  contours of pixels from the satellites are squares.  Is this a wrong assumption?

Reply:
Ahhhhh, I think I get it now. You get the corners of every individual pixels and the pixels are (of course) square. Then it makes perfect sense.

I think I got confused because you say "we are getting the upper left, upper right, lower left and lower right pixels", but what you mean is "we get the upper left, upper right, lower left and lower right coordinates of every pixel." I think that would be a valuable change!